version: '3.9'

services:
  mariadb:
    image: 'docker.io/bitnami/mariadb:latest'
    ports:
      - '3306:3306'
    environment:
      - "MARIADB_DATABASE=${DB_NAME}"
      - "MARIADB_USER=${DB_USER}"
      - "MARIADB_PASSWORD=${DB_PASSWORD}"
      - "MARIADB_ROOT_PASSWORD=${DB_ROOT_PASSWORD}"
    healthcheck:
        test: mysqladmin -p${DB_ROOT_PASSWORD} ping -h localhost
        timeout: 5s
        retries: 5
 
  privacyidea:
    build: .
    ports:
      - '8085:80'
    environment:
      - DB_VENDOR=mariadb
      - DB_HOST=mariadb
      - DB_NAME=${DB_NAME}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - SECRET_KEY=${SECRET_KEY}
      - PI_PEPPER=${PI_PEPPER}
      - PI_PAGE_TITLE=${PI_PAGE_TITLE}
      - PI_EXTERNAL_LINKS=True
      - PI_LOGFILE=${PI_LOGFILE}
      - PI_ADMIN_USER=${PI_ADMIN_USER}
      - PI_ADMIN_PASSWORD=${PI_ADMIN_PASSWORD}
    volumes:
      - ./certs:/etc/privacyidea/certs
    depends_on:
       mariadb:
          condition: service_healthy
